pkg/virt-controller/watch/vm_test.go

```diff
@@ -5335,6 +5335,64 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(vmConditionController.HasCondition(vm, virtv1.VirtualMachineRestartRequired)).To(BeTrue())
 				})
 			})
+
+			Context("Affinity", func() {
+				It("should work", func() {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								VMRolloutStrategy: &liveUpdate,
+								DeveloperConfiguration: &v1.DeveloperConfiguration{
+									FeatureGates: []string{virtconfig.VMLiveUpdateFeaturesGate},
+								},
+							},
+						},
+					})
+
+					vm, vmi := DefaultVirtualMachine(true)
+					addVirtualMachine(vm)
+
+					affinity := k8sv1.Affinity{
+						PodAffinity: &k8sv1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []k8sv1.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"test": "nmc",
+										},
+									},
+								},
+							},
+						},
+					}
+					vm.Spec.Template.Spec.Affinity = affinity.DeepCopy()
+					Expect(vmiInformer.GetIndexer().Add(vmi)).To(Not(HaveOccurred()))
```

Comment:
```suggestion
					Expect(vmiInformer.GetIndexer().Add(vmi)).To(Succeed())
```

p.s. (nit) personally I don't like these one-liner execute+validate. I think the code is much clearer when they're separate to different lines. It's also easier to debug that way. But there are other opinions on that :)

Reply:
That suggestion is important for readability. We don't want the VMI add to "not have occurred", we want it to succeed.
However I tend to disagree with the one-liner comment. "expect add vmi to succeed" is easier to read than "err = add vmi ; expect err to not have occurred" IMO.

Reply:
The one-liner `Succeed()` version is the preferred one indeed.

Reply:
Yeah I understand that I'm in the minority with this opinion :)